### PR TITLE
Initial Storybook Docs

### PR DESCRIPTION
**Pull request by danforbes**
Also renames WalletReceive component to WalletAddressHelpers

### CODE REVIEW

File: packages/circle-demo-webapp/app/components/WalletAddressHelpers/WalletAddressHelpers.tsx [hunk 1]
```diff
@@ -5,11 +5,16 @@ import type { Wallet } from 'web3-circle-sdk';
...
+  /** The wallet */
   wallet: Wallet;
 }
 
-export function WalletReceive({ wallet }: WalletReceiveProps) {
+/**
+ * Helpers for obtaining a wallet's on-chain address:
+ * a QR code that encodes the address and elements for viewing the address and copying it to the clipboard
+ */
+export function WalletAddressHelpers({ wallet }: WalletAddressHelpersProps) {
```
Comment (krzysu) [inline packages/circle-demo-webapp/app/components/WalletAddressHelpers/WalletAddressHelpers.tsx:17]: thinking about it, Helpers is not a good name for a component. A component represents something tangible. In this case could be just WalletAddress (probably not accurate as the component represents more) or WalletExtendedAddress or WalletAddressGroup or WalletAddressDetails or WalletReceiveDetails (from the purpose of the component) or WalletCopyableAddress (from the function). You get the idea 😄

File: packages/circle-demo-webapp/app/routes/wallet.$id/components/WalletAddressDialog.tsx [hunk 2]
```diff
@@ -4,25 +4,25 @@ import { Wallet } from 'web3-circle-sdk';
...
-export function WalletReceiveDialog({ wallet }: WalletReceiveDialogProps) {
+export function WalletAddressDialog({ wallet }: WalletAddressDialogProps) {
   const [open, setOpen] = useState(false);
 
   return (
     <Dialog open={open} onOpenChange={setOpen}>
       <DialogTrigger asChild>
         <Button variant="outline">
-          <ArrowDown /> Receive
+          <ArrowDown /> Address
```
Comment (krzysu) [inline packages/circle-demo-webapp/app/routes/wallet.$id/components/WalletAddressDialog.tsx:20]: The button Receive is adapted from the design so I'd not change that 
Screenshot 2024-12-13 at 09 38 36

File: packages/circle-demo-webapp/app/routes/wallet.$id/components/WalletAddressDialog.tsx [hunk 3]
```diff
@@ -4,25 +4,25 @@ import { Wallet } from 'web3-circle-sdk';
...
-import { WalletReceive } from '~/components/WalletReceive';
+import { WalletAddressHelpers } from '~/components/WalletAddressHelpers';
 
-interface WalletReceiveDialogProps {
+interface WalletAddressDialogProps {
   wallet: Wallet;
 }
 
-export function WalletReceiveDialog({ wallet }: WalletReceiveDialogProps) {
+export function WalletAddressDialog({ wallet }: WalletAddressDialogProps) {
```
Comment (krzysu) [inline packages/circle-demo-webapp/app/routes/wallet.$id/components/WalletAddressDialog.tsx:13]: I think here you went to far with the renaming as the reusable component name is one thing (as explained in my other comment) and the context it is used is another thing. And in this case the component is used in WalletReceiveDialog as it's purpose is display wallet details to receive funds. The name is also connected to the label of the button that triggers the feature. wdyt?

**Review by krzysu — CHANGES_REQUESTED** [review]
@danforbes in the individual comments I tried to explain a mindset that I have as a frontend dev when naming components, let me know if that makes sense for you. The requested change is only in the naming of the components and Receive button label. The rest is great. 🙂

File: packages/circle-demo-webapp/app/components/WalletAddressHelpers/WalletAddressHelpers.tsx (hunk 1 above)
Comment (danforbes) [reply packages/circle-demo-webapp/app/components/WalletAddressHelpers/WalletAddressHelpers.tsx]: Got it! I just reverted the rename for now 💯

File: packages/circle-demo-webapp/app/routes/wallet.$id/components/WalletAddressDialog.tsx (hunk 3 above)
Comment (danforbes) [reply packages/circle-demo-webapp/app/routes/wallet.$id/components/WalletAddressDialog.tsx]: All good...reverted 💯

**Review by krzysu — APPROVED** [review]

**Pull request merged**